### DISCUSSION TITLE
Inconsistent behavior between GetCompareExchangeValueOperation and GetCompareExchangeValuesOperation (1 and N items)

### DISCUSSION THREAD

**Discussion by georgiosd** [ravendb/ravendb#14751] [General]
Hey all :)

This seems a little off - am I missing anything?

![image](https://user-images.githubusercontent.com/1672054/184710964-4654d468-02b4-4273-920a-12d5b039dd74.png)

GetCompareExchangeValueOperation seems fine but I would have expected that GetCompareExchangeValuesOperation would always return a dictionary with the keys requested (doesn't do so for 1 key - middle case) and that the value for not found keys would be null instead of a result with `Value = null`.

Alas, I guess it could be a breaking change for clients that are already using it this way.

**Comment by ayende** [comment]
You are correct, the issue is here:

https://github.com/ravendb/ravendb/blob/aae47c57ca8fe01683690a6d484bb2f1882552e0/src/Raven.Server/Web/System/CompareExchangeHandler.cs#L88

**Reply by ayende** [reply]
The root issue is that we _used_ to have separate endpoint (See: https://github.com/ravendb/ravendb/commit/b1d36a251789626ea46d02ddfb415dc8a600d07f )

We merged them, and kept the single key behavior of returning `null`.

**Comment by georgiosd** [comment]
Should I open an issue or do you think it'll be a problem for existing users?

**Reply by ayende** [reply]
I wonder at what level to fix that?
Because `GetCompareExchangeValueOperation` _should_ return `null`, and we currently don't have a way to distinguish between one and many at the wire level

**Reply by georgiosd** [reply]
The client can re-create the dictionary from the keys passed as arguments?

**Reply by ayende** [reply]
Yes, of course. In fact, the client can just return empty doc if it gets null response.
